cnd/src/respawn.rs

```diff
@@ -0,0 +1,110 @@
+//! This module deals with respawning swaps upon startup of cnd.
+//!
+//! "Respawning" spawns refers to the feature of _spawning_ tasks into a runtime
+//! for watching the necessary ledgers of all swaps that we know about in the
+//! database which have not been completed yet.
+
+use crate::{
+    db::{Load, LoadAll},
+    swap_protocols::{halight, herc20, Facade, LocalSwapId, Side},
+};
+use chrono::Utc;
+use comit::{
+    lnd::{LndConnectorAsReceiver, LndConnectorAsSender},
+    Protocol, Role,
+};
+use tokio::runtime::Handle;
+
+/// Describes a swap that needs to be respawned.
+///
+/// We define this as a new type within this module instead of reusing a
+/// different one so we can semantically differentiate between swaps that need
+/// to be re-spawned and simply "all" swaps.
+#[derive(Debug)]
+pub struct Swap<A, B> {
+    pub id: LocalSwapId,
+    pub role: Role,
+    pub alpha: A,
+    pub beta: B,
+}
+
+/// Respawn the protocols for all swaps that are not yet done.
+pub async fn respawn(facade: Facade, handle: Handle) -> anyhow::Result<()> {
+    let swaps = facade.load_all().await?;
+
+    for swap in swaps {
+        match (swap, &facade.lnd_connector_params) {
+            (
+                Swap {
+                    id,
+                    role,
+                    alpha: Protocol::Herc20,
+                    beta: Protocol::Halight,
+                },
+                Some(lnd_params),
+            ) => {
+                let swap: Swap<herc20::Params, halight::Params> = match facade.load(id).await {
```

Comment:
I thought we prefer turbofish over explicit type?

Reply:
I prefer it over typing `Load::load(facade, ...`.